forte/data/data_store.py

```diff
@@ -490,6 +492,72 @@ def get_entry(self, tid: int) -> Tuple[List, int, int]:
         # Otherwise, use ``index_id`` to find the index of the entry.
         raise NotImplementedError
 
+    def co_iterator(self, type_names: List[str]) -> Iterator[List]:
+        r"""Given two or more type names, iterate their entry lists from beginning to end together.
+
+        For every single type, their entry lists are sorted by the begin and
+        end fields. The co_iterator function will iterate those sorted lists
+        together, and yield each entry in sorted order. This tasks is quite
+        similar to merging several sorted list to one sorted list. If two
+        entries have both the same begin and end field, then their order is
+        decided by the order of user input type_name (the type that first
+        appears in the target type list will return first).
+
+        Args:
+            type_names (List[str]): a list of string type names
+
+        Returns:
+
+            An iterator of entry elements.
+        """
+
+        n = len(type_names)
+        # suppose the length of type_names is N and the length of entry list of
+        # one type is M
+        # then the time complexity of using min-heap to iterate
+        # is O(M*log(N))
+
+        entries = []
+        for tn in type_names:
+            if tn in self.__elements:
+                entries.append(self.__elements[tn][0])
```

Comment:
We will probably use some constants defined here: https://github.com/asyml/forte/pull/635/files#diff-bed4b362357669e7f878c5c84fde9f9b6ed8f1a1d88b81a3e143a547ac8fa9c1, probably blocked until then.

Reply:
The constants for entry information are here: https://github.com/asyml/forte/blob/master/forte/common/constants.py

If you didn't find the const you need, you can consider add one.

Reply:
The constants for entry information are here: https://github.com/asyml/forte/blob/master/forte/common/constants.py

If you didn't find the const you need, you can consider add one.